AirSkyBoat/AirSkyBoat
# Ranged Distance Correction Implementation | Sharpshot Exception | PUP Automaton Ranged Attack Placeholder

**_I affirm:_**

## What does this pull request do?
+ Adds distance correction to ranged attack and ranged accuracy for players and automatons.
+ Players distance uses https://ffxiclopedia.fandom.com/wiki/Distance?oldid=1076603 as a source.
+ Automatons until further work can be performed use a mock formula that does not punish the master if the pet is too close.
+ Added packet updates to allow for the new RATT and RACC function params.
+ Added an exception for Sharpshot in the RACC formula to avoid the distance correction per https://ffxiclopedia.fandom.com/wiki/Sharpshot?oldid=1042098 .

## Steps to test these changes
+ Ran all distance modifier calcs through excel to ensure accuracy.
+ Tested if the distance did change the values of RACC and RATT.
+ Tested to see if Sharpshot removed the RACC penalty from being too close.
